server/src/test/java/org/elasticsearch/action/ActionModuleTests.java

```diff
@@ -286,7 +288,45 @@ public void test3rdPartyHandlerIsNotInstalled() {
                 e.getMessage(),
                 Matchers.equalTo(
                     "The org.elasticsearch.action.ActionModuleTests$SecPlugin plugin tried to "
-                        + "install a custom REST interceptor. This functionality is not available anymore."
+                        + "install a custom REST interceptor. This functionality is not available to external plugins."
+                )
+            );
+        } finally {
+            threadPool.shutdown();
+        }
+    }
+
+    public void test3rdPartyRestControllerIsNotInstalled() {
```

Comment:
can this really happen? we only export the `org.elasticsearch.plugins.interceptor` to serverless and security.
Someone would have to run ES server without java9 modules? (tests don't have them enabled)

Reply:
Yes, it's possible to have a plugin that is not a module, in which case qualified exports are not respected.

Perhaps we need to get to a place where we require that plugins are modularized, but we don't today.

Reply:
I think that if a plugin is not a module it would compile, but it would fail at runtime
I would expect that we would get a runtime exception along the lines
```
Exception in thread "main" java.lang.IllegalAccessError: superinterface check failed: class wrong.plugin.XYZ (in unnamed module @0x19469ea2) cannot access class org.elasticsearch.plugins.interceptor.RestServerActionPlugin (in module org.elasticserach.server) because module org.elasticsearch.server does not export org.elasticsearch.plugins.interceptor to unnamed module @0x19469ea2
```
(note the above message I 'hand crafted' basing on a simple reproduction project structure with 4 classes)

at the same time I don't mind this test, I just thought we don't need to wory about this (I don't test for other internal plugins)

Reply:
TIL.
I thought I had tested with a non-modularized plugin, but I suspect it might have worked because I temporarily had an open export.